csi/node_server.go

```diff
@@ -126,7 +126,11 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	if volume.State != string(longhorn.VolumeStateAttached) || volume.Controllers[0].Endpoint == "" {
 		log.Infof("Volume %v hasn't been attached yet, unmounting potential mount point %v", volumeID, targetPath)
 		if err := unmount(targetPath, mounter); err != nil {
-			log.WithError(err).Warn("Failed to unmount")
+			log.WithError(err).Warnf("Failed to unmount: %v", targetPath)
+		}
+		log.Infof("Volume %v hasn't been attached yet, unmounting potential mount point %v", volumeID, stagingPath)
+		if err := unmount(stagingPath, mounter); err != nil {
+			log.WithError(err).Warnf("Failed to unmount %v", stagingPath)
```

Comment:
Do we really need to unmount here? If the stagingPath is corrupted, I think the NodePublishVolume() call already handle it, right?

Reply:
Yes, removed it. Thanks